devito/types/basic.py

```diff
@@ -889,7 +699,12 @@ def __new__(cls, *args, **kwargs):
         args, kwargs = cls.__args_setup__(*args, **kwargs)
 
         # Extract the `indices`, as perhaps they're explicitly provided
-        dimensions, indices = cls.__indices_setup__(*args, **kwargs)
+        dim_args = cls.__indices_setup__(*args, **kwargs)
+        try:
```

Comment:
why not letting `__indices_setup__` do all this tinkering?

Reply:
That would require to add a bunch of it in all the Array/TempFunction/.... that don't need to know about staggered.